pkg/rancher-components/src/components/StringList/StringList.vue

```diff
@@ -360,6 +356,47 @@ export default Vue.extend({
       }
     },
 
+    /**
+     * Add a new or update an exiting item in the items list.
+     *
+     * @param items The current items list.
+     * @param value The new value to be added.
+     * @param index The list index of the item to be updated (optional).
+     * @returns Updated items list.
+     */
+    addValueToItems(items: string[], value: string, index?: number): string[] {
```

Comment:
I think the `items` parameter may be decoupled from the return value here:
```
const newItems = [...items];
...
return newItems;
```
and change the input wherever `addValueToItems` is called: `addValueToItems([...this.items]) -> addValueToItems(this.items)`